kliff/dataset/dataset.py

```diff
@@ -320,17 +510,63 @@ def __init__(
         self,
         path: Optional[Path] = None,
         weight: Optional[Weight] = None,
-        file_format="xyz",
+        file_format: str = "xyz",
```

Comment:
Now that we have multiple ways to create a dataset, It would be good to do it using individual staticmethod, instead of the if else statement. The major drawback of this if else statement approach is the mix of incompatible arguments together. What happens if a user passes `path` and `colabfit_dataset` together? We will have to do checks in __init__. 

If would propose the below:
```python
def __init__(self, configurations: list[Configuration]):
    self.configs = configurations

@staticmethod
def from_files(cls, path, <other args>):
     # read configs from the file(s) and assign to self.configs

@staticmethod
def from_colabfit(cls, database, <other args>):
     # query configs and assign to self.configs
```

Then the `add_configs` function needs to be updated as well, probably split into two? `add_from_files` and `add_from_colabfit`? 

The actual code to read the files and query colabfit can be shared, e.g. between `from_files` and `add_from_files`.

Reply:
Added new methods, `from_*` and `add_from_*` for colabfit, path, and ase based datasources. See comment below.